crates/telio-sockets/src/protector/apple.rs

```diff
@@ -245,6 +245,8 @@ impl Protector for NativeProtector {
             socks.notify.notify_waiters();
         }
     }
+
+    fn set_fwmark(&self, _fwmark: u32) {}
```

Comment:
What was the reason behind unifying the trait between the platforms? Just to get rid of `cfg`s in multiple places? Maybe we can use `unimplemented!();` to make it clear that those should not be used on particular platforms?

Reply:
Yes, I decided that we don't really gain anything by not having one or two methods on some systems. It's not that they shouldn't be used - the just have an empty implementation. I think that's pretty normal for an interface to have a no-op as an implementation for some of its methods and I would prefer to avoid something that can theoretically panic at runtime when called. Can we agree on some doc comments for those 'new' methods that will explain that currently they have no effect?

Reply:
Yes, I thought  mostly about documentation value when proposing `unimplemented!();` and if you prefer comments to be on the safe side I'm fine with those

Reply:
Comments added :)